src/backend/joanie/core/api/admin/__init__.py

```diff
@@ -398,6 +404,77 @@ def destroy(self, request, *args, **kwargs):
                 status=HTTPStatus.FORBIDDEN,
             )
 
+    @extend_schema(
+        request=None,
+        responses={
+            (200, "application/json"): OpenApiTypes.OBJECT,
+            404: serializers.ErrorResponseSerializer,
+        },
+    )
+    @action(
+        methods=["GET"],
+        detail=True,
+    )
+    def check_certificates_generation_process(self, request, pk=None):  # pylint:disable=unused-argument
+        """
+        Checks if the certificates generation task by Celery is ongoing or completed.
+        If cache data is found, generation is ongoing, else, it's completed or
+        it has not been requested.
+        """
+        course_product_relation = self.get_object()
+
+        if cache_data := cache.get(
+            f"celery_certificate_generation_{course_product_relation.id}"
+        ):
+            return JsonResponse(cache_data, status=HTTPStatus.OK)
+
+        return JsonResponse(
+            {"details": "No cache data found."}, status=HTTPStatus.NOT_FOUND
+        )
+
+    @extend_schema(
+        request=None,
+        responses={
+            (201, "application/json"): OpenApiTypes.OBJECT,
+            (202, "application/json"): OpenApiTypes.OBJECT,
+            400: serializers.ErrorResponseSerializer,
+        },
+    )
+    @action(
+        methods=["POST"],
+        detail=True,
+    )
+    def generate_certificates(self, request, pk=None):  # pylint:disable=unused-argument
+        """
+        Generate the certificates for a course product relation when it is eligible.
+        """
+        course_product_relation = self.get_object()
+        cache_key = f"celery_certificate_generation_{course_product_relation.id}"
+
+        if cache_data := cache.get(cache_key):
+            return JsonResponse(cache_data, status=HTTPStatus.ACCEPTED)
+
+        # Prepare cache data before trigger celery's task.
+        orders_ids = get_orders(course_product_relation)
+        certificates_published = get_generated_certificates(course_product_relation)
+        cache_data = {
+            "in_progress": True,
```

Comment:
IMO `in_progess` property is useless. Indeed, the fact that something exists in the cache is telling that a task is in progress by itself.